src/coreclr/vm/finalizerthread.cpp

```diff
@@ -580,6 +592,15 @@ void FinalizerThread::FinalizerThreadWait()
 
         DWORD status;
         status = hEventFinalizerDone->Wait(INFINITE,TRUE);
+
+        if (desiredFullGcCount - g_fullGcCountSeenByFinalization > 0)
```

Comment:
I assume that the subtraction and > 0 are to handle GC counts wrapping around. Signed integer overflows are undefined behavior in C++. I am not sure whether it can bite us here. Unsigned integer overflows have behavior in C++, so it would be more portable to do this with unsigned types.

Reply:
Yes, a billion of gen2 gc would be a rare, but plausible scenario for a long term service. 
I’d prefer the API to return size_t or uint64_t here, but it is what it is. Not sure why signed int was chosen.

We do know it is a monotonically growing count that may wrap around. I’ll throw in some unsigned casts, just in case the compiler does something unexpected with compares.

Reply:
One billion seconds is about 30 years. ;)

1-10 times per second is the rough guess for gen2 frequency, I think, excluding stress scenarios.